javascript/ql/src/semmle/javascript/frameworks/EventEmitter.qll

```diff
@@ -0,0 +1,225 @@
+import javascript
+
+module EventEmitter {
+  /** Gets the name of a method on `EventEmitter` that returns `this`. */
+  string chainableMethod() {
+    result = "off" or
+    result = "removeAllListeners" or
+    result = "removeListener" or
+    result = "setMaxListeners" or
+    result = on()
+  }
+
+  /** Gets the name of a method on `EventEmitter` that registers an event handler. */
+  string on() {
+    result = "addListener" or
+    result = "on" or
+    result = "once" or
+    result = "prependListener" or
+    result = "prependOnceListener"
+  }
+
+  /**
+   * Gets a node that refers to an EventEmitter object.
+   */
+  DataFlow::SourceNode trackEventEmitter(EventEmitter::Range emitter) {
```

Comment:
We are in the module already, so this can be simplified (this is just a stylistic choice).
```suggestion
  DataFlow::SourceNode trackEventEmitter(Range emitter) {
```

Reply:
Ditto elsewhere in this module..

Reply:
Perhaps this should even be: 

```suggestion
  DataFlow::SourceNode trackEventEmitter(EventEmitter emitter) {
```